### PR TITLE
Added serialization from etcd error metric

### PR DESCRIPTION
**Pull request by micahhausler**
**What type of PR is this?**

/kind bug
/sig api-machinery 

**What this PR does / why we need it**:

This is in response to #69579 and from [sig-api-machinery discussion on 3/11/20](https://docs.google.com/document/d/1x9RNaaysyO0gXHIr1y50QFbiL1x8OWnk2v3XnrdkT5Y/edit#heading=h.q8o43vubc18t).

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Added etcd serialization error metric
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

N/A

### CODE REVIEW

**Comment by wenjiaswe** [comment]
/lgtm
cc @jingyih who was on  #69579, @logicalhan who is assigned as mentor in [sig-api-machinery discussion on 3/11/20](https://docs.google.com/document/d/1x9RNaaysyO0gXHIr1y50QFbiL1x8OWnk2v3XnrdkT5Y/edit#heading=h.q8o43vubc18t).

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go [hunk 1]
```diff
@@ -137,7 +138,12 @@ func (s *store) Get(ctx context.Context, key string, resourceVersion string, out
 		return storage.NewInternalError(err.Error())
 	}
 
-	return decode(s.codec, s.versioner, data, out, kv.ModRevision)
+	err = decode(s.codec, s.versioner, data, out, kv.ModRevision)
+	if err != nil {
+		metrics.RecordDecodeError(typeName)
+		klog.V(4).Infof("Decoding %s \"%s\" failed", typeName, key)
```
Comment (jingyih) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go:144]: Let's also log the actual error.

**Review by jingyih — COMMENTED** [review]
Thank you! This looks good. Added one comment.

**Pull request closed without merging**